addons/Omen/Omen.lua

```diff
@@ -0,0 +1,287 @@
+-- Copyright © 2017, Braden, Sechs
+-- All rights reserved.
+-- 
+-- Redistribution and use in source and binary forms, with or without
+-- modification, are permitted provided that the following conditions are met:
+-- 
+--     * Redistributions of source code must retain the above copyright
+--       notice, this list of conditions and the following disclaimer.
+--     * Redistributions in binary form must reproduce the above copyright
+--       notice, this list of conditions and the following disclaimer in the
+--       documentation and/or other materials provided with the distribution.
+--     * Neither the name of Omen nor the
+--       names of its contributors may be used to endorse or promote products
+--       derived from this software without specific prior written permission.
+-- 
+-- THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+-- ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+-- WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+-- DISCLAIMED. IN NO EVENT SHALL Braden OR Sechs BE LIABLE FOR ANY
+-- DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+-- (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+-- LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+-- ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+-- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+-- SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+_addon.name    = 'Omen'
+_addon.author  = 'Braden, Sechs'
+_addon.version = '1.3'
+_addon.command = 'omen'
+
+config = require ('config')
+texts = require('texts')
+
+defaults = T{}
+defaults.text_R = 255 --Color values are in RGB, ranging from 0 to 255
+defaults.text_G = 255
+defaults.text_B = 255
+defaults.good_R = 0
+defaults.good_G = 255
+defaults.good_B = 0
+defaults.bad_R = 255
+defaults.bad_G = 0
+defaults.bad_B = 0
+defaults.pos_x = 0
+defaults.pos_y = 0
+defaults.font_size = 11
+
+settings = config.load(defaults)
+
+good_col = "\\cs("..tostring(settings.good_R)..","..tostring(settings.good_G)..","..tostring(settings.good_B)..")"
+bad_col = "\\cs("..tostring(settings.bad_R)..","..tostring(settings.bad_G)..","..tostring(settings.bad_B)..")"
+omens = 0
+obj_time = 0
+floor_obj = "Waiting for objectives..."
+floor_clear = ""
+IMAGE = texts.new("IMAGE")
```

Comment:
This is a nice idea, but it's saving a lot of stuff that shouldn't be into the xml, I mean A LOT. Any way to reduce that? I fear it might be a bit overwhelming for someone opening and trying to customize things, plus I would have to define more defaults at the beginning of the addon. (I already intended to add the BG alpha among the settings, anyway)